lib/backpex/live_resource.ex

```diff
@@ -1231,6 +447,803 @@ defmodule Backpex.LiveResource do
+
+    {:ok, socket}
+  end
+
+  defp assign_active_fields(socket, session) do
+    fields =
+      socket.assigns.live_resource.fields()
+      |> filtered_fields_by_action(socket.assigns, :index)
+
+    saved_fields = get_in(session, ["backpex", "column_toggle", "#{socket.assigns.live_resource}"]) || %{}
+
+    active_fields =
+      Enum.map(fields, fn {name, %{label: label}} ->
+        {name,
+         %{
+           active: field_active?(name, saved_fields),
+           label: label
+         }}
+      end)
+
+    socket
+    |> assign(:active_fields, active_fields)
+  end
+
+  defp assign_metrics_visibility(socket, session) do
+    value = get_in(session, ["backpex", "metric_visibility"]) || %{}
+
+    socket
+    |> assign(metric_visibility: value)
+  end
+
+  defp assign_filters_changed_status(socket, params) do
+    %{assigns: %{live_action: live_action}} = socket
+
+    socket
+    |> assign(:filters_changed, live_action == :index and params["filters_changed"] == "true")
+  end
+
+  defp field_active?(name, saved_fields) do
+    case Map.get(saved_fields, Atom.to_string(name)) do
+      "true" ->
+        true
+
+      "false" ->
+        false
+
+      _other ->
+        true
+    end
+  end
+
+  def assign_items(socket) do
+    %{
+      live_resource: live_resource,
+      fields: fields
+    } = socket.assigns
+
+    criteria = build_criteria(socket.assigns)
+    items = Resource.list(fields, socket.assigns, live_resource, criteria)
+
+    assign(socket, :items, items)
+  end
+
+  defp maybe_assign_metrics(socket) do
+    %{
+      assigns:
+        %{
+          repo: repo,
+          schema: schema,
+          live_action: live_action,
+          live_resource: live_resource,
+          fields: fields,
+          query_options: query_options,
+          metric_visibility: metric_visibility
+        } = assigns
+    } = socket
+
+    filters = active_filters(assigns)
+
+    metrics =
+      socket.assigns.live_resource.metrics()
+      |> Enum.map(fn {key, metric} ->
+        query =
+          EctoAdapter.list_query(
+            assigns,
+            &socket.assigns.live_resource.item_query(&1, live_action, assigns),
+            fields,
+            search: search_options(query_options, fields, schema),
+            filters: filter_options(query_options, filters)
+          )
+
+        case Backpex.Metric.metrics_visible?(metric_visibility, live_resource) do
+          true ->
+            data =
+              query
+              |> Ecto.Query.exclude(:select)
+              |> Ecto.Query.exclude(:preload)
+              |> Ecto.Query.exclude(:group_by)
+              |> metric.module.query(metric.select, repo)
+
+            {key, Map.put(metric, :data, data)}
+
+          _visible ->
+            {key, metric}
+        end
+      end)
+
+    socket
+    |> assign(metrics: metrics)
+  end
+
+  @impl Phoenix.LiveView
+  def render(%{live_action: action} = assigns) when action in [:show, :show_edit] do
```

Comment:
```suggestion
  def render(%{live_action: action} = assigns) when action in [:show] do
```

What is `show_edit` even for? It is the only occurrence.